src/test/e2e-emulated/contracts/semantics.tact

```diff
@@ -0,0 +1,233 @@
+    self.a1 += 100;
+    self.a2.b2.c1 += 1;
+}
+
+contract SemanticsTester {
+
+    // Currently, declaring fields like this:
+    // 
+    // sC: SC = SC {c1: 5}; 
+    // sB: SB = SB {b1: true, b2: sC, b3: 10};
+    // sA: SA = SA {a1: 20, a2: sB};
+    //
+    // or like this:
+    //
+    // sC: SC = SC {c1: 5}; 
+    // sB: SB = SB {b1: true, b2: self.sC, b3: 10};
+    // sA: SA = SA {a1: 20, a2: self.sB};
+    //
+    // gives a compilation error. So, we need to define the fields like this:
+
+    sC: SC = SC {c1: 5}; 
+    sB: SB = SB {b1: true, b2: SC {c1: 5}, b3: 10};
+    sA: SA = SA {a1: 20, a2: SB {b1: true, b2: SC {c1: 5}, b3: 10}};
+    uB: SB;
+
+    // An alternative would be to assign them in the init() function, which allows the use of self.
+
+    init() { 
+        self.uB.b1 = false;       // What is the meaning of this? self.uB is not initialized.
+                                  // Is this instantiating a partially constructed struct SB?
+                                  // Nevertheless, it is not possible to check if it is a partially initialized
+                                  // struct because the compiler will not allow referencing self.uB in an expression
+                                  // until self.uB is actually assigned.
+        self.uB.b2 = SC {c1: 40};
+        self.uB = SB {b1: false, b2: SC {c1: 0}, b3: 11};
+        self.uB.b3 = 14;          // Getter structInitCheck later checks that init respects 
+                                  // this last assignment
+
+        self.sB.b1 = false;
+        self.sB.b2 = SC {c1: 3};
+    }
+
+    receive() { }
+
+    // IMPORTANT: The operator == does not allow comparing two structs.
+    // So, to compare if two structs are equal using ==, one needs to compare each
+    // field of both structs manually.
+
+
+    // Assigning a struct to a variable preserves fields
+
+    get fun structAssign1(): Bool { 
+        let s = self.sA;
+
+        // The fields of s and self.sA are equal, and the values did not change
+        return s.a1 == self.sA.a1 && 
+               s.a2.b1 == self.sA.a2.b1 &&
+               s.a2.b2.c1 == self.sA.a2.b2.c1 &&
+               s.a2.b3 == self.sA.a2.b3 &&
+               s.a1 == 20 && 
+               s.a2.b1 == true &&
+               s.a2.b2.c1 == 5 &&
+               s.a2.b3 == 10 &&
+               self.sA.a1 == 20 && 
+               self.sA.a2.b1 == true &&
+               self.sA.a2.b2.c1 == 5 &&
+               self.sA.a2.b3 == 10;
+    }
+
+    // Assignment of structs is done by value
+
+    get fun structAssign2(): Bool { 
+        // Make a local copy of the struct
+        let s = self.sA;
+
+        // Modify two fields in the local copy
+        s.a1 = 50;
+        s.a2.b2.c1 = 70;
+
+        // Make a copy of an internal struct of s
+        let t = s.a2;
+
+        // Modify the copy
+        t.b3 = 100;
+        
+        // self.sA remains unchanged,
```

Comment:
Let's replace here and in all the places below the code that checks `self.sA` is unchanged with a helper function `checkSAUnchanged` or something like this, otherwise one has to check all these

```
self.sA.a1 == 20 && 
self.sA.a2.b1 == true &&
self.sA.a2.b2.c1 == 5 &&
self.sA.a2.b3 == 10
```